components/VideoGrid.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <div class="video-grid__wrapper">
+    <v-text-field
+      class="video-grid__search"
+      :label="`Search in all ${videos.length} videos`"
+      v-model="search"
+      @input="handleQueryChange"
+      single-line
+      hide-details
+      prepend-icon="mdi-magnify"
+    />
+    <div class="video-grid__list">
+      <VideoGridItem
+        v-for="video in paginatedVideos"
+        :key="video.id"
+        :video="video"
+        :highlighted-term="search"
+        @click.native="$router.push(`/watch/${video.id}`)"
+      />
+    </div>
+    <div class="text-center mt-5">
+      <v-pagination
+        :key="search"
+        color="green"
+        v-model="page"
+        :length="pagesCount"
+        :total-visible="7"
+      ></v-pagination>
+    </div>
+  </div>
+</template>
+
+<script>
+  import VideoGridItem from "~/components/VideoGridItem";
+
+  const PAGE_SIZE = 25
+
+  const isInName = (item, search) => RegExp(search, 'i').test(item.name)
+  const isInDescription = (item, search) => RegExp(search, 'i').test(item.description)
+  const isInCodeSummary = (item, search) => RegExp(search, 'i').test(item.code_summary)
+  const isInCourseName = (item, search) => RegExp(search, 'i').test(item.courseName)
+
+  export default {
+    name: 'VideoGrid',
+    components: {VideoGridItem},
+    data(){
+      return {
+        search: '',
+        page: 1
+      }
+    },
+
+    mounted() {
+      if (this.routeHasSearchQuery) {
+        this.search = this.$route.query.q
+      }
+    },
+    computed: {
+      pagesCount() {
+        return Math.ceil(this.sortedFilteredVideos.length / PAGE_SIZE)
+      },
+      filteredVideos() {
+        if(this.search.length === 0) { return this.videos }
+        return this.videos.filter(item => isInName(item, this.search) || isInDescription(item, this.search) || isInCodeSummary(item, this.search))
+      },
+      sortedFilteredVideos() {
+        return this.filteredVideos.sort((a, b) => {
+          let weightA = 0
+          let weightB = 0
+          if (isInName(a, this.search)) weightA += 2
```

Comment:
Great idea on weighting the sorting 👍 👍 👍

Reply:
I might add some weighting based on how long ago they were released, as well.  Not sure the exact formula on this yet.

Reply:
@jeffreybiles I tried to come up with a working and efficient formula, but nothing worthwhile came to mind, except the option to highlight and add some pointes to recent videos. I added a bonus weight and a corresponding icon for the videos released during the last week.
<img width="686" alt="Vue Screencasts - Intermediate and Advanced Tutorials, Trainings, and Mentorship 2020-05-10 21-54-08" src="https://user-images.githubusercontent.com/4090500/81502713-b6422b80-9309-11ea-8b18-d1107c9c80ee.png">